### ISSUE TITLE
Reposing Humans by Warping 3D Features

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#13]
![image](https://user-images.githubusercontent.com/46308407/104117619-f4469200-5365-11eb-831c-ddcca54fa566.png)

# Key Idea

- Implicitly learn **dense feature volume** (volumetric representation) from human images
- Conduct **geometric warping** in the feature space and map them back to RGB space by convolutional decoder.

# Previous Approaches

- Fully convolutional neural networks

    → Tend to depend strongly on pixels near the center of receptive field

    → Fail to move information across long distance

    → Difficulty in reposing as it requires long-distance manipulation

- Mesh based approaches

    → Fail when shape differs (ex) different hair style etc

# Formulation

- Two stream Generator Network
    1. Reposing 
    2. Inpainting

![image](https://user-images.githubusercontent.com/46308407/104117628-03c5db00-5366-11eb-878e-2672ed6b6110.png)

- Lifting Encoder
2D image → 3D volumetric feature
![image](https://user-images.githubusercontent.com/46308407/104117638-21934000-5366-11eb-88f1-940887cb0d81.png)

Reshaping Operation →
![image](https://user-images.githubusercontent.com/46308407/104117640-2e179880-5366-11eb-816b-9fd7613a2009.png)

- Warping Module

Take volumetric feature $V$ as well as $P_I, P_T$, the input and target Poses

Use $P_I$, the input pose to create 10 masks, one for each body part distinguished by joints

Clone $V$ 10 times and apply mask by voxel-wise multiplication and fit Transformation $T_i$

![image](https://user-images.githubusercontent.com/46308407/104117651-42f42c00-5366-11eb-8ffc-1b3704ebef2a.png)

← Why *maximum* activation?

Then the **Target Pose Encoder i**nputs Gaussian heatmap $H \in R^{H \times W \times D \times J}$ and the output is concatenated to volumetric features and processed by projection decoder. ($J$: # joints)

- **Projection Decoder**

    Mirrors the lifting encoder. 

    3D convolution $D_{3D}$ first learns the volume features and combines target pose encoding.

    Combines Depth and Channel dimension back together by reshaping. 

    Feed the 2D representation to 2D convolution, $D_{2D}$, yielding 2D image in RGB space.

- **Background Inpainter**

    Since the warping module masks body parts and copy them to decoder, the background info is lost. 

    Pixels not included in projected body part masks become part of background mask, $M_{BG}$

    Perform PartialConv operation and alpha blending (background and person) $\hat{M_{BG}}$ as weight.

- ResNet base, bottleneck residual blocks for 2D convs, and no bottlenecks for 3D convs.

- Objective Function

![image](https://user-images.githubusercontent.com/46308407/104117660-57d0bf80-5366-11eb-95ce-a56cc24d6a56.png)

$L_{perc}$: L1 loss on multiple generated/target image feature maps from ImageNet pretrained VGG

$L_{adv}$: Adversarial Loss. (Generated, input, 3D heatmap), (Ground truth, input, 3D heatmap) given

# Evaluation Metric

1. Structural Similarity Index (SSIM)
    - Compares patches of generated images and ground truth images
2. Learned Perceptual Image Patch Similarity (LPIPS)
    - Compares deep features as in $L_{perc}$
